tools/rewriter/SourceRewriter.cpp

```diff
@@ -1421,6 +1421,10 @@ int main(int argc, const char **argv) {
       wrapper_out << "asm(\n";
       wrapper_out << asm_wrapper;
       wrapper_out << ");\n";
+    } else {
+      header_out << "asm(\n";
+      header_out << "  \".set " << wrapper_name << ", __real_" << fn_name << "\\n\"\n";
```

Comment:
I think this branch is dead code and the TODO on line 1413 should be deleted. I mean if you take the address of a static function that runs in compartment 0 then you must always be in compartment 0 (since it's static) and the rewriter doesn't add annotations to source files in compartment 0, right?

Reply:
also if it isn't dead code wouldn't we want to add this in `wrapper_out` instead of the header?

Reply:
This isn't for static functions, this is for taking the address of a function in compartment 0 from compartment 1. The new ABI test has an example of this, I needed this to get that to even compile.

Reply:
`ret_in_memory` is in compartment 0, so for us to call `IA2_FN(ret_in_memory)` we need a wrapper, but the wrapper is a no-op so we just need to provide a symbol alias.

Reply:
sorry I was building the wrong targets when I went to check if it was dead code (still getting the hang of things again). That all makes sense. Also having it in `header_out` is fine since that gives simpler control flow than having a no-op call gate in the callgates .so and it shouldn't be an issue that it's in a header since the symbol name already can't be duplicated in different translation units.